### ISSUE TITLE
Requires macOS version 10.15

### ISSUE THREAD

**Issue by homeclubber** [jeffvli/feishin#534]
V6.0 seems to require macOS version 10.15, i cant see any mention of this in the changelog.

I'm on 10.13 and would like to update.

**Comment by BobWs** [comment]
Same here I want to try this app in combo with navidrome but I can't install it in macOS 10.13 (High Sierra)!

**Comment by jeffvli** [comment]
What's the actual error that you're getting when trying to install? I bumped the electron version in the newest version but I didn't see any breaking changes regarding macOS version compatibility.

**Comment by homeclubber** [comment]
I get the following error why i try to run the package on 10.13.6

![Untitled](https://github.com/jeffvli/feishin/assets/74516342/8ca01d50-59e4-497d-b8b2-13b083dbbd69)

If i go into the package and run the executable i get the following report. 

```
Process:               Feishin [4636]
Path:                  /Applications/Feishin 2.app/Contents/MacOS/Feishin
Identifier:            Feishin
Version:               0.6.0 (0.6.0)
Code Type:             X86-64 (Native)
Parent Process:        bash [4597]
Responsible:           Feishin [4636]
User ID:               501

Date/Time:             2024-03-12 08:56:52.043 +0000
OS Version:            Mac OS X 10.13.6 (17G14042)
Report Version:        12
Anonymous UUID:        3D0476A7-9226-3220-54A6-B6AE0715458D

Sleep/Wake UUID:       17D8AEBE-E24C-447B-B58E-886E7C294B30

Time Awake Since Boot: 16000 seconds
Time Since Wake:       74 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    DYLD, [0x1] Library missing

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: /System/Library/Frameworks/AuthenticationServices.framework/Versions/A/AuthenticationServices
  Referenced from: /Applications/Feishin 2.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework
  Reason: image not found

Binary Images:
       0x10a448000 -        0x10a4498ab +org.jeffvli.feishin (0.6.0 - 0.6.0) <4C4C442B-5555-3144-A1CE-CB43F8717240> /Applications/Feishin 2.app/Contents/MacOS/Feishin
       0x111763000 -        0x1117adadf  dyld (551.5) <CB9BFB56-4511-36F1-A546-891FF770C01C> /usr/lib/dyld
       0x111801000 -        0x11a86ecd7 +com.github.Electron.framework (27.1.0) <4C4C4430-5555-3144-A1BE-16E7A44C5BD9> /Applications/Feishin 2.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework
    0x7fff36270000 -     0x7fff36277fff  com.apple.IOSurface (211.15 - 211.15) <08DA3232-805F-304B-80DD-C86B49CC4AAD> /System/Library/Frameworks/IOSurface.framework/Versions/A/IOSurface
    0x7fff59689000 -     0x7fff5968affb  libSystem.B.dylib (1252.50.4) <5A02016C-8F9D-3F47-8C39-AD2CD4F9F11D> /usr/lib/libSystem.B.dylib

Model: MacBookPro8,1, BootROM 87.0.0.0.0, 2 processors, Intel Core i7, 2.8 GHz, 16 GB, SMC 1.68f99
Graphics: Intel HD Graphics 3000, Intel HD Graphics 3000, Built-In
Memory Module: BANK 0/DIMM0, 8 GB, DDR3, 1333 MHz, 0x859B, 0x43543130323436344246313333392E4D3136
Memory Module: BANK 1/DIMM0, 8 GB, DDR3, 1333 MHz, 0x802C, 0x31364B544631473634485A2D314734443120
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0xD6), Broadcom BCM43xx 1.0 (5.106.98.102.30)
Bluetooth: Version 6.0.7f22, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en1
Serial ATA Device: Patriot P210 128GB, 128.04 GB
Serial ATA Device: APPLE HDD HTS547575A9E384, 750.16 GB
USB Device: USB 2.0 Bus
USB Device: Hub
USB Device: IR Receiver
USB Device: USB 2.0 Bus
USB Device: FaceTime HD Camera (Built-in)
USB Device: Hub
USB Device: Apple Internal Keyboard / Trackpad
USB Device: BRCM2070 Hub
USB Device: Bluetooth USB Host Controller
Thunderbolt Bus: MacBook Pro, Apple Inc., 25.1

```

**Comment by jeffvli** [comment]
Got it, seems like I missed this breaking change on Electron v27.

[Removed: macOS 10.13 / 10.14 support](https://www.electronjs.org/blog/electron-27-0#removed-macos-1013--1014-support). I'll build a new release to fix this tomorrow.

**Comment by BobWs** [comment]
Same as @homeclubber ...

**Comment by BobWs** [comment]
I'm sorry but it still isn't working...keep getting the not compatible message (see screenshot)
I even tried to change the minimumVersion in the `info.plist` file but that didn't help.

![Screen_Shot](https://github.com/jeffvli/feishin/assets/5485820/33ee526b-4bb6-4223-8a10-70ccad5488b4)

**Comment by homeclubber** [comment]
same, running the executable gives the following

```
$ /Volumes/Feishin\ 0.6.1/Feishin.app/Contents/MacOS/Feishin ; exit;
dyld: Library not loaded: /System/Library/Frameworks/AuthenticationServices.framework/Versions/A/AuthenticationServices
  Referenced from: /Volumes/Feishin 0.6.1/Feishin.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework
  Reason: image not found
Abort trap: 6
logout
```

**Comment by jeffvli** [comment]
Is anyone familiar with npm?

Try re-packaging the app yourselves while downgrading the electron version until it works. I don't have a macOS device to test which version in v26 actually works. Feishin used to use v25 but it was causing breakage in NixOS which is why it was upgraded to v27.

Tldr is:
1. Clone/download the repo
2. Make sure nodejs/npm is installed
3. Check [electron v26 versions](https://www.npmjs.com/package/electron?activeTab=versions)
4. Inside `package.json` file in the root directory, update the `electron` package version under `devDependencies` to a previous version
5. From bash/console, navigate to the root of the repo folder and run  `npm install` **(this needs to be run every time you change  the electron version)**
6. Run `npm run package` and wait for the app to build
7. Try running the new build inside the `./release/build` folder

**Comment by homeclubber** [comment]
Could we go the other way and update the electron version?

**Comment by jeffvli** [comment]
From the Electron v27 changelog that I posted in my first comment, it documents that macOS 10.14 and 10.13 support was dropped in v27. But considering it still doesn't work in the latest version of v26 means that you need to downgrade.

I know for sure that Electron v25 works, but we need to find out which version if any v26 will work.

**Comment by BobWs** [comment]
Is is possible to just clone a version for macOS 10.13.* and Electron v25, and keep the rest on Electron v25. 
This way you only need to keep one version for the older macOS and no need to keep it up-to-date with the latest Electron versions. 
And for all the other users with newer macOS you can use the Electron v27 and up...
Just my 2 cents...

**Comment by kgarner7** [comment]
Latest Feishin uses electron 31. If you wish to maintain compatibility with older systems, please see https://github.com/jeffvli/feishin/discussions/691#discussioncomment-10215778 for necessary changes
